### PR TITLE
frontend/feat/#2868/firebase_realtime_database_support : retrieval of…

### PR DESCRIPTION
**Pull request by Anuragini-2506**
… static data for kiosk locations from firebase

## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Retrieval of kiosk lcoation data from firebase realtime database

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?

Tested locally

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 1]
```diff
@@ -253,6 +261,27 @@ private static HashMap<String, String> getQueryMap(String link) {
...
     }
 
+    private void getFirebaseRealTimeDatabase(){
+        final FirebaseDatabase database = FirebaseDatabase.getInstance();
+        DatabaseReference ref = database.getReference();
+        // Kiosk locations
+        ref.child("location").addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                ArrayList<KioskLocation> kioskLocations =  (ArrayList<KioskLocation>) dataSnapshot.getValue();
```
Comment (Vignesh-772) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:271]: getting warnings in this line

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 2]
```diff
@@ -253,6 +261,27 @@ private static HashMap<String, String> getQueryMap(String link) {
         return query_params;
     }
 
+    private void getFirebaseRealTimeDatabase(){
+        final FirebaseDatabase database = FirebaseDatabase.getInstance();
+        DatabaseReference ref = database.getReference();
+        // Kiosk locations
+        ref.child("location").addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
```
Comment (Vignesh-772) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:270]: annotate with `@NonNull`

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 3]
```diff
@@ -253,6 +261,27 @@ private static HashMap<String, String> getQueryMap(String link) {
         return query_params;
     }
 
+    private void getFirebaseRealTimeDatabase(){
```
Comment (Vignesh-772) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:264]: if possible try to move it to mobility-driver module

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]
resolve comments

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 4]
```diff
@@ -253,6 +261,27 @@ private static HashMap<String, String> getQueryMap(String link) {
...
+        DatabaseReference ref = database.getReference();
+        // Kiosk locations
+        ref.child("location").addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                ArrayList<KioskLocation> kioskLocations =  (ArrayList<KioskLocation>) dataSnapshot.getValue();
+                JSONArray jsArray = new JSONArray(kioskLocations);
+                SharedPreferences sharedPref = getApplicationContext().getSharedPreferences(
+                        activity.getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+                sharedPref.edit().putString("KIOSK_LOCATIONS", "" + jsArray).apply();
```
Comment (Vignesh-772) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:275]: use `jsArray.toString()`

**Comment by witcher-shailesh** [comment]
Not required, as of now. Could be replaced with RemoteConfigs in future

**Pull request closed without merging**